app/client/src/components/pages/WaterbodyReport.js

```diff
@@ -614,14 +615,38 @@ function WaterbodyReport({ fullscreen }: Props) {
 
         setWaterbodyUses({ status: 'success', data: uses });
 
-        const sources = probableSources.map((source) => {
+        const data = [];
+        probableSources.forEach((source) => {
           const name = source.sourceName;
           const status = source.sourceConfirmedIndicator === 'N' ? 'No' : 'Yes';
-
-          return { name, status };
+          source.associatedCauseNames.forEach((cause) => {
+            for (let parameter of parameters) {
+              if (parameter.parameterName === cause.causeName) {
+                const use =
+                  parameter.associatedUses.length &&
+                  parameter.associatedUses[0].associatedUseName;
+                data.push({
+                  source: titleCaseWithExceptions(name),
+                  parameter: titleCaseWithExceptions(cause.causeName),
+                  confirmed: status,
+                  use: use,
+                });
+                break;
+              }
+            }
+          });
         });
 
-        setWaterbodySources({ status: 'success', data: sources });
+        data
+          .sort((a, b) => a.source > b.source)
+          .sort((a, b) => {
+            if (a.source === b.source) {
+              return a.parameter > b.parameter;
+            }
+            return -1;
+          });
```

Comment:
```suggestion
        data
          .sort((a, b) => b.source.localeCompare(a.source))
          .sort((a, b) => {
            if (a.source === b.source) {
              return a.parameter.localeCompare(b.parameter);
            }
            return -1;
          });
```

This sort isn't working as I would expect it to. Currently, the sorting is sorting by string length instead of in alphabetical order. This suggestion will sort in alphabetical order and it should also fix the code smells flagged by sonar cloud.

You can use the following as an example of the issue: http://localhost:3000/waterbody-report/DOEE/DCANA00E_01/2020

Reply:
Good call! Also, I didn't realize that's a builtin function, cool.